app/models/custom_button_set.rb

```diff
@@ -1,6 +1,13 @@
 class CustomButtonSet < ApplicationRecord
   acts_as_miq_set
 
+  before_save :update_children
+  def update_children
+    replace_children Rbac.filtered(CustomButton.where(:id => set_data[:button_order]))
```

Comment:
Let's imagine a hypothetical scenario where you have 2 users (A and B). User A has access to the buttons x,y but not to z while user B has access to the buttons y, z but not to x. There's a custom button set already available with buttons x and z. What happens if either user A or user B tries to add to this set the button y? I think if user A touches this piece of code, the button set will contain x,y but not z even if it was originally in the set. Same goes for user B but then we will loose button x. Not sure if this situation is possible, if yes then we have a "loss of precision" here.

Reply:
Agreed, you're right. (I also have no idea how custom button sets act for different users, or indeed if rbac over custombuttons does anything. Anybody?)

Maybe we should just push RBAC filtering to the loading side, and skip it when saving?

Reply:
The practical implication of the PR as it stands now (with `Rbac.filtered`) is: a user won't be able to edit a custom button set if it contains any buttons that user can't see.

Reply:
I am not able to find any RBAC rule for `CustomButton` model. Are you referencing to the rule
"User has access to some CustomButtons" according to `CustomButton#userid` ? 

if there is existence of such rule, it seems that it is not implemented in `Rbac.filtered` call.

So should I implement it to the RBAC ?

Reply:
> So should I implement it to the RBAC ?

No, it already does that (because the sets will not be the same after filtering, so if will abort).

If there are currently no rbac rules for custom buttons, that's fine, that probably means we don't have a problem at all, right @skateman ?

(Though it makes sense to make the solution correct for when rbac gets involved anyway.)

Reply:
@himdel yup, who knows when this can :boom: into our face when RBAC gets smarter :confused:

Reply:
:+1: so, which behaviour makes most sense then?

We can allow a user to edit sets where they can't access all the buttons, and not remove those buttons. But that means a user can theoretically add or remove buttons they don't have access to.

Or we can only allow a user to edit a set where that user has access to all the buttons. But then, it means they can potentially do less than they could do now. (Well, when there is a rule to make some buttons inaccessible to that user anyway.)